### DISCUSSION TITLE
Seamless Online Join

### DISCUSSION THREAD

**Discussion by tryintomakeit** [YimMenu/YimMenu#1529] [Ideas]
as seen on the fork https://github.com/TupoyeMenu/TupoyeMenu by @tupoy-ya , something that really caught my attention, there's an option that essentially makes it so you can seamless join online lobbies without the sky view loading screen ever triggering, allowing you to essentially keep playing on the map while joining an online session and spawn whenever you were at the moment in singleplayer

![options](https://github.com/YimMenu/YimMenu/assets/137688030/3b11600d-c899-40db-91d4-eccbd13f6652)

https://github.com/YimMenu/YimMenu/assets/137688030/39eeeb4b-c910-40c6-9ade-fb2ad8bdee9a

**Comment by Tayipc38** [comment]
[Tupoye's code](https://github.com/TupoyeMenu/TupoyeMenu/blob/b157662001a71f73d7c8737d2deab00219576940/src/backend/looped/tunables/seamless_join.cpp#L11)

**Reply by tupoy-ya** [reply]
No it's just the skycam stopping code, the real code is in `src/native_hooks/maintransition.hpp`.

**Reply by Tayipc38** [reply]
Ah. Seems like something to yoink right out from ur base.

**Reply by tupoy-ya** [reply]
I don't think it's finished yet, i want it to keep the vehicle you are in and not play the intro animation, as well test for all kinds of infinite loadings it can cause(they are quite rare now if your spawn point is "last location"*).

Sadly i'm doing some useless stuff instead of actually finishing it.

**Reply by Tayipc38** [reply]
The task sequence for the intro animation is done in freemode.

**Comment by binded-zz** [comment]
i love that menu for the window layouts and menu docking to what ever side of the window you want 

there is some bugs with that for me   the infinate loading that he just talked about 
ex. flying in a sparrow and changing session you will fall out the heli and smash in to the ground and then load loop 
i end up using the bail network to solo offline then rejoin a new session 
idk if the same happens if your in a car or what ever else 
he did fix some of that with a report that i had made and its much better then it was before
